code/modules/vehicles/unmanned/remote_control.dm

```diff
@@ -0,0 +1,48 @@
+/datum/component/remote_control
+	var/mob/user
+	var/atom/target
+	var/is_controlling = FALSE
+
+
+/datum/component/remote_control/Initialize(atom/target)
+	. = ..()
+	src.target = target
+
+	RegisterSignal(parent, COMSIG_REMOTE_CONTROL_TOGGLE, .proc/toggle_remote_control)
+
+/datum/component/remote_control/Destroy()
+	return ..()
+
+/datum/component/remote_control/proc/on_user_clickon(datum/source, atom/A, params)
+	if(A==parent)
+		return ..()
+	if(istype(target, /obj/vehicle/unmanned))
+		var/obj/vehicle/unmanned/T = target
+		T.handle_remotecontrol_onuserclickon(A, params)
+
+	return COMSIG_MOB_CANCEL_CLICKON //So that the click returns there and doesn't go on.
+
+
+/datum/component/remote_control/proc/toggle_remote_control(datum/source, mob/user)
+	if(!is_controlling)
+		src.user = user
```

Comment:
Do this in `remote_control_on()`
It should probably never be turned on without a user.